File: www/cgi-bin/python/test_content_length.py
#!/usr/bin/env python3
import sys

# CGI output with explicit Content-Length header
body = "<html><body><h1>CGI with Content-Length</h1><p>This response includes a Content-Length header.</p></body></html>"
content_length = len(body)

print("Content-Type: text/html")
print("Content-Length: %d" % content_length)
print()
print(body)

File: www/cgi-bin/php/test_content_length.php
#!/usr/bin/env php-cgi
<?php
// Return a custom Content-Length header
$body = str_repeat("X", 100);
$content_length = 50; // Lie about length

header("Content-Type: text/plain");
header("Content-Length: $content_length");
echo $body; // Actually sends 100 bytes
?>
